px/px_commandline.py

```diff
@@ -319,6 +322,61 @@ def get_command(commandline: str) -> str:
     return app_name_prefix + command
 
 
+def get_guile_command(commandline: str) -> Optional[str]:
+    array = to_array(commandline)
+    array = list(filter(lambda s: s, array))
+
+    # Ignore some switches.
+    IGNORE_SWITCHES = [
+        "-s",
+        "--listen",
+        "-ds",
+        "--debug",
+        "--no-debug",
+        "--auto-compile",
+        "--fresh-auto-compile",
+        "--no-auto-compile",
+        "-q",
+        "--r6rs",
+        "--r7rs",
+        "-h",
+        "--help",
+        "-v",
+        "--version",
+    ]
+    IGNORE_ARGFUL_SWITCHES = [
+        "-L",
+        "-C",
+        "-x",
+        "-l",
+        "-e",
+    ]
+    while len(array) > 1 and (
+        array[1] in IGNORE_SWITCHES
+        or array[1] in IGNORE_ARGFUL_SWITCHES
+        or array[1].startswith("--language=")
+        or array[1].startswith("--listen=")
+        or array[1].startswith("--use-srfi=")
+    ):
+        if len(array) > 1 and array[1] in IGNORE_ARGFUL_SWITCHES:
```

Comment:
The length is guaranteed by the surrounding while loop.

```suggestion
        if array[1] in IGNORE_ARGFUL_SWITCHES:
```